src/Symfony/Component/Security/Http/Tests/Fixtures/CustomUser.php

```diff
@@ -44,6 +45,15 @@ public function eraseCredentials(): void
 
     public function __serialize(): array
     {
-        return [\sprintf("\0%s\0username", self::class) => $this->username];
+        $data = (array) $this;
+        $passwordKey = \sprintf("\0%s\0password", self::class);
+
+        if ($this->hashPassword) {
+            $data[$passwordKey] = hash('crc32c', $this->password);
+        } else {
+            unset($data[$passwordKey]);
```

Comment:
Previously, the private field was not unset. I wonder if you're no longer testing the case where the password hash is serialized as is.

Reply:
case added